### PR TITLE
realvnc-vnc-viewer: 7.5.1 -> 7.12.0 && move to by-name/

### PR DESCRIPTION
**Pull request by jackyliu16**
## Description of changes

realvnc-vnc-viewer: 7.5.1 -> 7.12.0 
move to by-name/

### **NOTE:** Suspected destructive problem 
I can't tell if the package runs on darwin(to poor to have one), by running this have a kind of error like bellow and I don't know if this is BC i run it in incorrect platform or it's dependence issue, 
- I test via add some package like darwin.apple_sdk.frameworks.CoreFoundation to buildInputs, but would not help with this situation.

```
➜  nixpkgs git:(update-realvnc) ✗ nix-build . -A realvnc-vnc-viewer --option system x86_64-darwin
this derivation will be built:
  /nix/store/xmqd5xpkz17mzdycgfn5fmiv35a1xsb9-realvnc-vnc-viewer-7.12.0.drv
error: derivation '/nix/store/xmqd5xpkz17mzdycgfn5fmiv35a1xsb9-realvnc-vnc-viewer-7.12.0.drv' requested impure path '/System/Library/Frameworks/CoreFoundation.framework', but it was not in allowed-impure-host-deps
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/re/realvnc-vnc-viewer/darwin.nix [hunk 1]
```diff
@@ -12,7 +12,7 @@ stdenvNoCC.mkDerivation (finalAttrs: {
   src = fetchurl rec {
       name = "VNC-Viewer-${finalAttrs.version}-MacOSX-universal.dmg";
       url = "https://downloads.realvnc.com/download/file/viewer.files/${name}";
-      sha256 = "0k72fdnx1zmyi9z5n3lazc7s70gcddxq0s73akp0al0y9hzq9prh";
+      sha256 = "1i9mly54qqxhiy6z9p0q8px3n1rc014vdxjzsmn3mx25q11gd845";
```
Comment (Aleksanaa) [inline pkgs/by-name/re/realvnc-vnc-viewer/darwin.nix:15]:
```suggestion
      hash = "";
```

Use SRI hash instead (you have to regenerate it)

Comment (jackyliu16) [reply pkgs/by-name/re/realvnc-vnc-viewer/darwin.nix]: regenerate via `nix hash to-sri --type sha256`

**Comment by jackyliu16** [comment]
@NixOS/nixpkgs-merge-bot merge

**Comment by Lassulus** [comment]
Result of `nixpkgs-review pr 323949` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>realvnc-vnc-viewer</li>
  </ul>
</details>

**Pull request merged**